### DISCUSSION TITLE
Calling for testers

### DISCUSSION THREAD

**Discussion by Elliria** [autokey/autokey#888] [General]
Hey there,

I've been putting together the steps for testing any of the AutoKey code (clones, releases, etc.) in a virtual environment (like **venv**, which is what I used) since a virtual environment is required when using [pip](https://pypi.org/project/pip/) commands in Debian and Ubuntu 23.04 and beyond. 

I've created a [GitHub Gist that shows how it's done](https://gist.github.com/Elliria/d881fa2fc0fb2083b15b166f7fb8feee) that contains the steps and I'd love it if any of you would be willing to follow those steps and let me know if they work or don't work for you or if any of them are unclear or missing or incorrect. If they're solid, they can go into the wiki.

**Comment by Elliria** [comment]
Suggested label: Development

**Reply by josephj11** [reply]
Anything like this needs to be posted on Gitter and on Google Groups. It probably won't be seen by many people here.

**Comment by Elliria** [comment]
Okay, thanks for the reminder. I hardly ever visit Gitter any more (really do not enjoy the new platform), but I put the call for testers on there now.

**Reply by josephj11** [reply]
I'm not in love with Gitter, but it does support markdown which is quite helpful.

They're still working on the unread messages indicators problems.

You might want to take another look at our channel on Discord and see if you like that any better. I believe their smartphone apps are better too. We could consider making that more primary. I believe it also offers voice and video chats as well which might be useful/fun.

**Reply by Elliria** [reply]
Gitter supporting Markdown is definitely a plus, especially when copying stuff over from the AutoKey project.

They don't seem to genuinely believe us that there's an issue when it comes to the message indicators, or, more accurately, they don't seem to understand exactly what's wrong with the indicators when we explain it.

They also edit our bug reports to the point where they're completely unrecognizable and have nothing to do with what we initially reported, yet leave our names on the reports [sigh]. Here's hoping that despite all of that, they'll manage to fix it.

I took a look at Discord when Gitter first changed over and I found Discord difficult to use.

As for the phones, that's something I wouldn't know about. I only use my smartphone as a phone and don't add apps to it.

**Comment by josephj11** [comment]
The latest comments in the Gitter room indicate that they fixed a problem that prevented new messsages from updating the counts. Once that was done, it revealed several bugs that we're dewaling with now. Those are being actively worked on, so it may be resolved sometime soon, but no target dates have been mentioned.

**Reply by Elliria** [reply]
I just logged on to find that my response to someone complaining about those notifications got deleted. Also, when reviewing issue reports that I've submitted or contributed to, I've been either pretty well or completely obliterated. It's not a very encouraging landscape to find oneself on.

**Comment by josephj11** [comment]
Discord is easy to use, but it has a common fatal flaw - when you reply to a message, your new post goes at the end and so does your pointer, so you lose your unread/next message position. IDK why getting real threads to work is such a difficult thing to do, but only Matrix (and its element and gitter front ends) seems to know how to do that.

I blew away my whole account when trying to add a new channel to Discord - and lost access to a couple of rooms that are hard to get invited back into. So I definitely don't love it!

**Reply by Elliria** [reply]
Oh, my. The post and pointer going to the end of the messages so you lose your place would be a show-stopper for me.

One thing I like about several IRC clients I've used and currently use is that they put this little line across the screen at exactly the point you last were when you were in the current room. That way it's easy to tell what you haven't read yet. Maybe these chat programs should borrow some pages from the IRC playbooks.
